File: paramentrorefval.cpp
#include <iostream>
using namespace std;


class Pessoa
	{
	public:
	
	int idade;
	string nome;
	
	
	void Inicialisador(string n, int nu)
		{
		nome = n;
		idade = nu;	
		}
	
	
	
	};

	
void AlteraC(Pessoa A)
	{
	A.idade = 360;
	A.nome = "JOJO";
	}
	
void AlteraC(Pessoa* A)
	{
	A->idade = 360;
	A->nome = "JOJO";
	}

	
int main()
	{
	Pessoa P; 
	
	P.Inicialisador("Jhonatan", 100);  
	cout << P.idade<<" "<< P.nome<<"\n";
	
	AlteraC(P);
	cout << P.idade<<" "<< P.nome<<"\n";
	
	AlteraC(&P);
	cout << P.idade<<" "<< P.nome<<"\n";
	
	return(0);
	}
	
	

File: paramentrorefval.cs
using System;
					
public class Program
{
	public static void Main(String [] agrs)
	{
		Pessoa A = new Pessoa(100 ,"Jhonatan");
		Console.WriteLine(A.Idade + " " +A.Nome);
		
		MudaPessoa(A);
		Console.WriteLine(A.Idade + " " +A.Nome);
			
	}
	
	public static void MudaPessoa(Pessoa Pes)
		{
		Pes.Idade = 360;
		Pes.Nome = "JOJO";
	
		}
		
}


public class Pessoa
	{
	public int Idade;
	public string Nome;

	public Pessoa(int A, string N)
		{
		Idade = A;
		Nome = N;
		}
	}